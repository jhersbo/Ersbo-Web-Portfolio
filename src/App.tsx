import './sass/styles.scss'
import HeroNav from './components/HeroNav';
import LandingScreen from './components/LandingScreen';
import About from './components/About';

import { ReactNode } from 'react';
import React from 'react';

interface IMyProps{
  page: string
}
class App extends React.Component<{}, IMyProps>{
  constructor(props: any){
    super(props);
    this.state = {
      page: 'landing',
    }
    this.setView = this.setView.bind(this)
  }
  setView = (page: string)=>{
    this.setState({page: page})
  }
  render(): ReactNode {
    if(this.state.page === 'landing'){
      return(
        <div className='app'>
          <LandingScreen viewChange={this.setView}/>
        </div>
      )
    }else{
      return(
        <div className='app'>
          <HeroNav/>
        </div>
      )
    }
  }
}

export default App;