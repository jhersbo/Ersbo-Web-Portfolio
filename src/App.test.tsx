import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'

test('passes snapshot test', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
});
