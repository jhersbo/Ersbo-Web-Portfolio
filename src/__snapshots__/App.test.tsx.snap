// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot test 1`] = `
<div
  className="app"
>
  <div
    onClick={[Function]}
    onMouseUp={[Function]}
  >
    <header
      id="top-landing-panels"
    >
      <div
        className="top-landing-panels"
        id="landing-name-div"
        style={Object {}}
      >
        <h1
          id="landing-name"
          style={Object {}}
        >
          Jack
          <br />
          Ersbo
        </h1>
      </div>
      <div
        className="top-landing-panels"
        id="landing-role-div"
        style={Object {}}
      >
        <h1
          id="landing-role"
          style={Object {}}
        >
          Full-Stack Software Developer
        </h1>
      </div>
    </header>
    <footer
      className="bottom-landing-panel"
      id="landing-footer"
      style={Object {}}
    >
      <h2>
        Thank you for checking out my site!
      </h2>
      <h4
        id="landing-click-prompt"
      >
        *Click anywhere to continue*
      </h4>
    </footer>
  </div>
</div>
`;
